pkg/source/source_test.go

```diff
@@ -247,13 +248,24 @@ var _ = Describe("Source", func() {
 		It("should return an error if syncing fails", func(done Done) {
 			f := false
 			instance := source.NewKindWithCache(nil, &informertest.FakeInformers{Synced: &f})
-			err := instance.WaitForSync(nil)
+			err := instance.WaitForSync(context.Background(), time.Duration(10)*time.Second)
```

Comment:
Have added a test case - where the caches do not sync and because of the timeout which is set there is an error (this verifies that we don't go into an infinite loop waiting for caches to sync). However, verifying the case with a reasonable timeout requires us to pass a context to `controller.Start(..)` and wait for the channel to close. Currently, the [`fakeinformer` ](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/informertest/fake_cache.go#L83-L88) does not use the context and instead only mocks whether caches are synced or not to ensure that `Start()` is not blocked. I suppose that's why in all the other test cases we are passing a cancelled context. Modifying the fake informer to accept the context would break the other test cases. Is there a workaround for this, or is the current test sufficient?

Reply:
See my other comment, writing an integration test that uses a real cache and not the fakeinformer should be doable here

Reply:
Got it.. modified to include the test cases